src/instructlab/utils.py

```diff
@@ -564,3 +595,53 @@ def split_hostport(hostport: str) -> tuple[str, int]:
     if not hostname or not port:
         raise ValueError(f"Invalid host-port string: '{hostport}'")
     return hostname, port
+
+
+def convert_messages_to_legacy_dataset(
+    dataset: List[MessageSample],
+) -> List[LegacyMessageSample]:
+    """
+    Converts the new HuggingFace messages dataset format to the legacy format.
+    **Note**: The legacy dataset format assumes only a turn of 1. All extra turns will be dropped.
+    """
+    converted_dataset: List[LegacyMessageSample] = []
+    for dp in dataset:
+        # in new dataset, the roles of a message will be determined.
+        if len(dp["messages"]) < 3:
+            raise ValueError(
+                "Dataset cannot be missing messages from roles. Please make sure that a message from each role is present."
+            )
+        system = dp["messages"][0]["content"]
+        user = dp["messages"][1]["content"]
+        assistant = dp["messages"][2]["content"]
+
+        converted_dataset.append(
+            {
+                "system": system,
+                "user": user,
+                "assistant": assistant,
+            }
+        )
+    return converted_dataset
+
+
+def is_messages_dataset(
```

Comment:
I think the function name could be clearer. Like  `get_messages_from_dataset()`?

Reply:
For a boolean function?

Reply:
@leseb Or are you talking about all of the functions overall? GitHub is just showing me your comment on `is_messages_dataset`.

Reply:
Argh, I missed that part. I guess it was too early for me. :). DIsmiss what I said.

Reply:
Okay, no worries.